File: 1968 array-with-elements-not-equal-to-average-of-neighbors/solution.go
func rearrangeArray(nums []int) []int {
    for i:=1;i<len(nums)-1;i++ {
            if 2*nums[i] == nums[i-1] + nums[i+1] {
                temp:= nums[i+1]
                nums[i+1] = nums[i]
                nums[i] = temp
                if i>1 {
                    i-=2
                } 
            }
        }
        return nums
}

File: 1968 array-with-elements-not-equal-to-average-of-neighbors/solution.cs
public class Solution {
    public int[] RearrangeArray(int[] nums) {
        for(int i=1;i<nums.Length-1;i++) {
            if(2*nums[i] == nums[i-1] + nums[i+1]) {
                int temp = nums[i+1];
                nums[i+1] = nums[i];
                nums[i] = temp;
                if(i>1) i-=2;
            }
        }
        return nums;
    }
}
